src/rules/strictBooleanExpressionsRule.ts

```diff
@@ -0,0 +1,116 @@
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+    public static BINARY_EXPRESSION_ERROR = "Invalid Expression: Operands for the && or || operator should be of type boolean";
+    public static UNARY_EXPRESSION_ERROR = "Invalid Expression: Operand for the ! operator should be of type boolean";
+    public static IF_STATEMENT_ERROR = "Invalid Expression: If statement condition needs to be a boolean expression or literal";
```

Comment:
Combine lines 7 to 11 into `Statement is not a boolean expression`. After that, combine some of the checker logic below to reduce code duplication.